datafusion/expr/src/expr_schema.rs

```diff
@@ -198,7 +199,8 @@ impl ExprSchemable for Expr {
             | Expr::IsNotTrue(_)
             | Expr::IsNotFalse(_)
             | Expr::IsNotUnknown(_)
-            | Expr::Exists { .. } => Ok(false),
+            | Expr::Exists { .. }
+            | Expr::Placeholder(_) => Ok(false), // todo: Placeholder should return false?
```

Comment:
I think since we don't know if the placeholder will be null, it should return true to be safe